src/util/entropyCreateStateFromJsons.ts

```diff
@@ -2,66 +2,96 @@ import { genotypeColors } from "./globals";
 import { defaultEntropyState } from "../reducers/entropy";
 
 type JsonAnnotations = Record<string, JsonAnnotation>
-// enum Strand {'+', '-'} // other GFF-valid options are '.' and '?'
-type Strand = string;
-type JsonSegmentRange = {start: number, end: number}; // Start is 1-based, End is 1-based closed (GFF)
+
+// '.' and '?' are GFF-valid options
+type Strand = '+' | '-' | '.' | '?'
+
+interface JsonSegmentRange {
+  /** 1-based */
+  start: number
+
+  /** 1-based closed (GFF) */
+  end: number
+}
+
 interface JsonAnnotation {
   /* Other properties are commonly set in the JSON structure, but the following are
   the only ones read by Auspice */
-  end?: number;
-  start?: number;
-  segments?: JsonSegmentRange[];
-  strand: Strand;
-  gene?: string;
-  color?: string;
-  display_name?: string;
-  description?: string;
+  end?: number
+  start?: number
+  segments?: JsonSegmentRange[]
+  strand?: Strand
```

Comment:
making sure this needed to become optional to match some usage somewhere?

Reply:
In 2d3a331c the matched usage was `Object.prototype.hasOwnProperty.call(annotation, "strand")` which to me indicates that it is optional. It's hard to tell from other usage because the entire `JsonAnnotation` object is represented by `json.meta.genome_annotations` here:

https://github.com/nextstrain/auspice/blob/a7578d20f1c4db7b3755ae501816e9c11967a6d8/src/actions/recomputeReduxState.js#L819

https://github.com/nextstrain/auspice/blob/a7578d20f1c4db7b3755ae501816e9c11967a6d8/src/actions/recomputeReduxState.js#L870

@jameshadfield might have a better idea.

Reply:
It's optional for the `nuc` annotation. You could argue it should be required, but we relaxed this and just assert that it's not negative:

https://github.com/nextstrain/auspice/blob/c0ca42fed852bcd2709dfa853e134cd7ca6f2f46/src/util/entropyCreateStateFromJsons.ts#L93-L94

For CDSs to be considered valid it must be "+" or "-":

https://github.com/nextstrain/auspice/blob/c0ca42fed852bcd2709dfa853e134cd7ca6f2f46/src/util/entropyCreateStateFromJsons.ts#L187-L188